src/usr/local/www/interfaces_vxlan_edit.php

```diff
@@ -0,0 +1,250 @@
+		$input_errors[] = gettext("VXLAN Local Port must be valid port value.");
+	}
+
+	if (!empty($_POST['vxlanremoteport']) && !is_port($_POST['vxlanremoteport'])) {
+		$input_errors[] = gettext("VXLAN Local Port must be valid port value.");
+	}
+
+	if (!empty($_POST['vxlanttl']) && (!is_numeric($_POST['vxlanttl']) ||
+	    ($_POST['vxlanttl'] < 1) || ($_POST['vxlanttl'] > 255))) {
+		$input_errors[] = gettext("VXLAN TTL must be in range between 1 and 255.");
+	}
+
+	foreach ($a_vxlans as $vxlan) {
+		if (isset($id) && ($a_vxlans[$id]) && ($a_vxlans[$id] === $vxlan)) {
+			continue;
+		}
+
+		if (($vxlan['if'] == $_POST['if']) && ($vxlan['remote-addr'] == $_POST['remote-addr']) &&
+		    ($vxlan['vxlanid'] == $vxlan['vxlanid'])) {
+			$input_errors[] = sprintf(gettext("A vxlan with the remote address %s and VXLAN ID %s is already defined."), $vxlan['remote-addr'], $vxlan['vxlanid']);
+			break;
+		}
+	}
+
+	if (!$input_errors) {
+		$vxlan = array();
+		list($vxlan['if'], $vxlan['ipaddr']) = explode("|", $_POST['if']);
+		$vxlan['remote-addr'] = $_POST['remote-addr'];
+		$vxlan['vxlanid'] = $_POST['vxlanid'];
+		$vxlan['vxlanlocalport'] = $_POST['vxlanlocalport'];
+		$vxlan['vxlanremoteport'] = $_POST['vxlanremoteport'];
+		$vxlan['vxlanttl'] = $_POST['vxlanttl'];
+		if (isset($_POST['vxlanlearn'])) {
+			$vxlan['vxlanlearn'] = '';
+		}
+		$vxlan['descr'] = $_POST['descr'];
+		$vxlan['vxlanif'] = $_POST['vxlanif'];
+		$vxlan['vxlanif'] = interface_vxlan_configure($vxlan);
+
+		if ($vxlan['vxlanif'] == "" || !stristr($vxlan['vxlanif'], "vxlan")) {
+			$input_errors[] = gettext("Error occurred creating interface, please retry.");
+		} else {
+			if (isset($id) && $a_vxlans[$id]) {
+				$a_vxlans[$id] = $vxlan;
+			} else {
+				$a_vxlans[] = $vxlan;
+			}
+
+			write_config();
+
+			$confif = convert_real_interface_to_friendly_interface_name($vxlan['vxlanif']);
+
+			if ($confif != "") {
+				interface_configure($confif);
+			}
+
+			header("Location: interfaces_vxlan.php");
+			exit;
+		}
+	}
+}
+
+function build_parent_list() {
+	$parentlist = array();
+	$portlist = get_possible_listen_ips();
+	foreach ($portlist as $ifn => $ifinfo) {
+		$parentlist[$ifn] = $ifinfo;
+	}
+
+	return($parentlist);
+}
+
+$pgtitle = array(gettext("Interfaces"), gettext("VXLANs"), gettext("Edit"));
+$pglinks = array("", "interfaces_vxlan.php", "@self");
+$shortcut_section = "interfaces";
+include("head.inc");
+
+if ($input_errors) {
+	print_input_errors($input_errors);
+}
+
+$form = new Form();
+
+$section = new Form_Section('VXLAN Configuration');
+
+$section->addInput(new Form_Select(
+	'if',
+	'*Parent Interface',
+	$pconfig['if'],
+	build_parent_list()
+))->setHelp('This interface serves as the local address to be used for the VXLAN tunnel.');
+
+$section->addInput(new Form_IpAddress(
+	'remote-addr',
+	'*VXLAN Remote Address',
+	$pconfig['remote-addr']
+))->setHelp('Peer address or multicast group where encapsulated vxlan packets will be sent.');
```

Comment:
VXLAN should probably be capitalized here since it is capitalized in the help text of other options.